src/js/forms.js

```diff
@@ -68,16 +56,34 @@ class Forms {
 	}
 
 	/**
-	 * Input validation
-	 * @param {Element} input - The input to validate
-	 */
-	validate(input) {
-		if (!input.validity.valid) {
-			input.closest('.o-forms-input').classList.add(this.className.invalid);
-			return false;
+	* Form validation
+	* Validates every element in the form
+	*/
+	validateForm () {
+		return this.formElements.map(input => input.validate());
+	}
+
+	/**
+	* Input state
+	* @param {String} [name] - name of the input fields to add state to
+	* @param {String} [state] - type of state to apply — one of 'saving', 'saved', 'none'
+	*/
+	setState(state, name) {
+		let object = this.stateArray.find(item => item.name === name);
+		if (!object) {
+			object = {
+				name,
+				element: new State(this.form.elements[name])
+			};
+
+			this.stateArray.push(object);
+		}
+
+		if (!state) {
+			throw new Error(`${state} is not a recognised state, the options are 'saving', 'saved' or 'none'.`);
```

Comment:
This error should live in `state.js`, so it throws here and if `state.js` is used directly.